applications/nrf_desktop/doc/dvfs.rst

```diff
@@ -0,0 +1,91 @@
+However, if performing more computation-consuming tasks DVFS module will dynamically change to a higher frequency to increase performance.
+
+The module is only available for nrf54h20 SoC.
+Available operating points (in documentation called frequency):
+
+1. DVFS_FREQ_HIGH - high frequency - 320 MHz, 0.8 V
+#. DVFS_FREQ_MEDLOW - medium-low frequency - 128 MHz, 0.6V
+#. DVFS_FREQ_LOW - low frequency - 64 MHz, 0.5V
+
+
+Module events
+*************
+
+.. include:: event_propagation.rst
+    :start-after: table_dvfs_start
+    :end-before: table_dvfs_end
+
+.. note::
+    |nrf_desktop_module_event_note|
+
+Configuration
+*************
+
+The Dynamic Voltage and Frequency Scaling (DVFS) module is enabled by default for nrf54h20 Soc and can be disabled by setting the :option:`CONFIG_DESKTOP_DVFS` Kconfig option to ``n``.
+
+
+The DVFS module can be configured using the following Kconfig options:
+
+1. :option:`CONFIG_DESKTOP_DVFS_RETRY_BUSY_TIMEOUT_MS` - Timeout in milliseconds specifying time after which DVFS module will retry DVFS frequency change.
+   This timeout is applied in case another DVFS change request is still in progress which causes the current request to fail.
+   The default value is 1 ms.
+#. :option:`CONFIG_DESKTOP_DVFS_RETRY_INIT_TIMEOUT_MS` - Timeout in milliseconds specifying time after which DVFS module will retry DVFS frequency change.
+   This timeout is applied in case DVFS is not yet initialized which causes the current request to fail.
+   The default value is 500 ms.
+#. :option:`CONFIG_DESKTOP_DVFS_RETRY_COUNT` - Number of retries of DVFS frequency change after which DVFS module will report MODULE_STATE_ERROR.
+   The default value is 5.
+
+Each DVFS state can be configured using the following Kconfig options:
+
+1. ``CONFIG_DESKTOP_DVFS_STATE_<STATE>_ENABLE`` - Enable DVFS state <STATE>.
+#. ``CONFIG_DESKTOP_DVFS_STATE_<STATE>_TIMEOUT_MS`` - Timeout in milliseconds specifying time after which the DVFS module returns to low frequency after entering <STATE>.
+#. ``CONFIG_DESKTOP_DVFS_STATE_<STATE>_ACTIVE_FREQ`` - Choice of active frequency of <STATE>.
+    The possible values are:  ``DESKTOP_DVFS_STATE_<STATE>_ACTIVE_FREQ_HIGH`` and ``DESKTOP_DVFS_STATE_<STATE>_ACTIVE_FREQ_MEDLOW``.
+
+Currently supported DVFS states are:
+
+1. ``DVFS_STATE_INITIALIZING`` - State indicating that core initialization is in process.
+#. ``DVFS_STATE_LLPM_CONNECTED`` - State indicating that Low Latency Packet Mode is active.
+#. ``DVFS_STATE_USB_CONNECTED`` - State indicating that USB is connected.
+#. ``DVFS_STATE_CONFIG_CHANNEL`` - State indicating that Config Channel is active.
+#. ``DVFS_STATE_SMP_TRANSFER`` - State indicating that SMP transfer is active.
+
+Implementation details
+**********************
+
+The Dynamic Voltage and Frequency Scaling (DVFS) has two state tracing entities.
```

Comment:
We have three different entities here and I would be careful with wording:

- the DVFS module (nRF Desktop module)
- the DVFS capability (HW feature)
- the DVFS service (part of nrfs to interact with the HW feature via SCFW)

Reply:
Hope I clarified it:
the DVFS module (nRF Desktop module) -  "nRF Desktop DVFS "
the DVFS capability (HW feature)  - "DVFS"
the DVFS service (part of nrfs to interact with the HW feature via SCFW) - "nrfs service DVFS"